resources/assets/js/snipeit_modals.js

```diff
@@ -122,11 +123,13 @@ $(function () {
                 $('#modal_error_msg').html(error_message).show();
                 return false;
             }
-            var id = result.payload.id;
-            var name = result.payload.name || (result.payload.first_name + " " + result.payload.last_name);
-            if(!id || !name) {
-                console.error("Could not find resulting name or ID from modal-create. Name: "+name+", id: "+id);
-                return false;
+            if(!hasnopayload) {
```

Comment:
I'm not sure why we'd need this? The modal is just a form post

Reply:
The note that is added doesn't have an id (it's not an actual modal) and the existing code expects the response to have an id (`var id = result.payload.id`). The `if` check allows getting around that chunk of expectations without re-writing too much of the existing functionality.

Reply:
It should not require any payload. It should post to the controller, same as files and every other modal (except for the dynamic ones, via the "new" buttons).

Reply:
You’re looking at the bespoke modal code - look at the regular file upload modal. There is no callback needed here. It’s just a regular form in a modal